src/main/kotlin/event/player/ItemFrameInteract.kt

```diff
@@ -0,0 +1,23 @@
+package event.player
+
+import org.bukkit.Material
+import org.bukkit.entity.Entity
+import org.bukkit.entity.ItemFrame
+import org.bukkit.event.player.PlayerInteractEntityEvent
+
+object ItemFrameInteract {
+
+    fun itemframeInteractEvent(event: PlayerInteractEntityEvent) {
+        val entity: Entity? = event.rightClicked
+
+        if (entity is ItemFrame) {
+            val itemFrame = entity
+
+            if (event.getPlayer().isSneaking() && event.getPlayer().getInventory().getItemInMainHand().getType() === Material.AIR) {
```

Comment:
Standard getters do not need to be explicitly called in kotlin so `event.player.isSneaking` is also valid. See also IDE suggestions.